src/Service/ServiceFactory.php

```diff
@@ -172,6 +172,19 @@ public function getProgrammesService(): ProgrammesService
         return $this->instances['ProgrammesService'];
     }
 
+    public function getPromotionsService(): PromotionsService
+    {
+        if (!isset($this->instances['PromotionsService'])) {
```

Comment:
Swapping these stringy cache keys with constants sounds like a good idea, but let's do that all at once in a new PR. We could do it for both the ServiceFactory and the MapperFactory as they both follow a similar pattern.

How about using the name of the class you're caching an instance of? e.g.

```php
    public function getSegmentsService(): SegmentsService
    {
        if (!isset($this->instances[SegmentsService::class])) {
            $this->instances[SegmentsService::class] = new SegmentsService(
                $this->entityManager->getRepository('ProgrammesPagesService:Segment'),
                $this->mapperFactory->getSegmentMapper(),
                $this->cache
            );
        }

        return $this->instances[SegmentsService::class];
    }
```